### PR TITLE
Update ToExpression to show float and decimal in compact/noncompact mode

### PR DESCRIPTION
**Pull request by nguhoa**
Fix issue: https://github.com/microsoft/Power-Fx/issues/1314

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Core.Tests/PrimitiveValueConversionsTests.cs [hunk 1]
```diff
@@ -37,21 +39,23 @@ public void TestAll(Type dotnetType, Type fxType)
...
                 var value = GetValue(dotnetType);
                 var fxValue = PrimitiveValueConversions.Marshal(value, dotnetType);
                 Assert.Equal(fxType, fxValue.Type.GetType());
 
                 var expr = actualFxType.DefaultExpressionValue();
-                var engine = new Engine(new PowerFxConfig());
+
+                SymbolTable symbol = new SymbolTable();
+                symbol.AddFunctions(BuiltinFunctionsCore._featureGateFunctions);
+                var engine = new Engine(new PowerFxConfig() { SymbolTable = symbol });
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/PrimitiveValueConversionsTests.cs:52]: nit: can we encapsulate this in a helper:
GetEngineWithFeatureGatedFunctions();

**Review by MikeStall — APPROVED** [review]
:shipit:

File: src/tests/Microsoft.PowerFx.Core.Tests/PrimitiveValueConversionsTests.cs (hunk 1 above)
Comment (nguhoa) [reply src/tests/Microsoft.PowerFx.Core.Tests/PrimitiveValueConversionsTests.cs:52]: yes, update the code

**Review by MikeStall — APPROVED** [review]
:shipit:

**Pull request merged**